--- a/eigen3.pc.in	2016-02-16 13:26:15.000000000 +0000
+++ b/eigen3.pc.in	2016-07-27 13:58:40.663907675 +0100
@@ -1,9 +1,8 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-exec_prefix=${prefix}
 
 Name: Eigen3
 Description: A C++ template library for linear algebra: vectors, matrices, and related algorithms
 Requires:
 Version: @EIGEN_VERSION_NUMBER@
 Libs:
-Cflags: -I${prefix}/@INCLUDE_INSTALL_DIR@
+Cflags: -I@INCLUDE_INSTALL_DIR@
